python/mpt/src/mpt/utils.py

```diff
@@ -0,0 +1,80 @@
+import logging
+
+from ethereum.cancun.trie import (
+    BranchNode,
+    ExtensionNode,
+    InternalNode,
+    LeafNode,
+    bytes_to_nibble_list,
+)
+from ethereum_rlp import rlp
+from ethereum_types.bytes import Bytes
+
+logger = logging.getLogger(__name__)
+
+
+def decode_node(node: Bytes) -> InternalNode:
+    """
+    Decode a node from its RLP encoding.
+
+    Parameters
+    ----------
+    node_data : Bytes
+        The RLP encoded node data
+
+    Returns
+    -------
+    InternalNode
+        The decoded node (BranchNode, ExtensionNode, or LeafNode)
+    """
+    """Decode an RLP encoded node into an InternalNode."""
```

Comment:
double docstring

+ I'm not in favor of keeping the parameters and returns description if it's obvious